cpp/src/arrow/csv/writer.cc

```diff
@@ -289,11 +306,23 @@ class CSVWriterImpl : public ipc::RecordBatchWriter {
       io::OutputStream* sink, std::shared_ptr<io::OutputStream> owned_sink,
       std::shared_ptr<Schema> schema, const WriteOptions& options) {
     RETURN_NOT_OK(options.Validate());
+    // Reject null string values that contain quotes.
+    if (CountEscapes(options.null_string) != 0) {
+      return Status::Invalid("CSV null value string cannot contain quotes.",
+                             options.null_string);
```

Comment:
This will result in a formatted message such as `CSV null value string cannot contain quotes.ab"c`, perhaps arrange it a bit better? Or we can simply omit the actual `null_string` value since it is a user-defined parameter anyway.